io-engine/src/core/volume_snapshot.rs

```diff
@@ -0,0 +1,46 @@
+use super::CoreError;
+use crate::{
+    lvs::{Lvol, LvolSnapshot},
+    subsys::NvmfReq,
+};
+use async_trait::async_trait;
+use spdk_rs::libspdk::spdk_bdev_io;
+
+///  Traits gives the common snapshot/clone interface for Local/Remote Lvol.
+#[async_trait(?Send)]
+pub trait VolumeSnapshotOps {
+    /// Create Snapshot Common API.
+    async fn create_snapshot(&self) -> Result<u64, CoreError>;
+
+    /// Return the list of Snapshots for Local/remote Device.
+    fn list_snapshot(&self, source_vol: &str) -> Vec<LvolSnapshot>;
+}
+pub trait VolumeSnapshotParams {
+    /// Get Transaction Id of the Snapshot Create.
+    fn get_txn_id(&self) -> Option<String>;
```

Comment:
nit: in rust getters tend not to be prefixed with "get_", so this could be come simply `txn_id`.
Now that we have a separate trait, we don't need to return Option<String>?
Simply this would do?
`fn get_txn_id(&self) -> String;`